src/vs/workbench/parts/terminal/electron-browser/terminalConfigHelper.ts

```diff
@@ -83,18 +100,60 @@ export class TerminalConfigHelper {
 		return DEFAULT_ANSI_COLORS[baseThemeId];
 	}
 
+	private initializeCharacterSize(font: ITerminalFont, parentDomElement: HTMLElement) {
+		if (!this.characterMeasureElement) {
+			this.characterMeasureElement = new Builder(parentDomElement, true).div().build().getHTMLElement();
+		}
+
+		this.characterMeasureElement.style.display = 'inline';
+		this.characterMeasureElement.style.fontFamily = font.fontFamily;
+		this.characterMeasureElement.style.fontSize = font.fontSize + 'px';
+		this.characterMeasureElement.style.lineHeight = font.lineHeight + 'px';
+		this.characterMeasureElement.innerText = 'X';
+		let bcr = this.characterMeasureElement.getBoundingClientRect();
+		this.characterMeasureElement.style.display = 'none';
+		this.charWidth = Math.ceil(bcr.width || TERMINAL_CHAR_WIDTH_FALLBACK);
+		this.charHeight = Math.ceil(bcr.height || TERMINAL_CHAR_HEIGHT_FALLBACK);
+	}
+
 	/**
-	 * Set the terminal font to `terminal.integrated.fontFamily` if it is set, otherwise fallback to
-	 * `editor.fontFamily`.
+	 * Gets the font information based on the terminal.integrated.fontFamily,
+	 * terminal.integrated.fontSize, terminal.integrated.lineHeight configuration properties
+	 *
+	 * @param parentDomElement Host for the character measurement element
+	 * @param updateCharSize A flag which indicates whether the character size should be measured
 	 */
-	public getFontFamily(): string {
+	public getFont(parentDomElement: HTMLElement, updateCharSize?: boolean): ITerminalFont {
```

Comment:
Can you change `updateCharSize?: boolean` to `measureCharSize: boolean = false`? That way you don't really need the jsdoc to clarify.